applications/GeoMechanicsApplication/tests/cpp_tests/test_utilities/spy_condition.h

```diff
@@ -0,0 +1,53 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#pragma once
+
+#include "includes/condition.h"
+
+namespace Kratos::Testing {
+
+class SpyCondition : public Condition {
+public:
+    void FinalizeSolutionStep(const ProcessInfo& rCurrentProcessInfo) override;
+    void InitializeSolutionStep(const ProcessInfo& rCurrentProcessInfo) override;
+    void InitializeNonLinearIteration(const ProcessInfo& rCurrentProcessInfo) override;
+    void FinalizeNonLinearIteration(const ProcessInfo& rCurrentProcessInfo) override;
+
+    void EquationIdVector(EquationIdVectorType& rResult,
+                          const ProcessInfo& rCurrentProcessInfo) const override;
+
+    void GetDofList(DofsVectorType& rElementalDofList,
+                    const ProcessInfo& rCurrentProcessInfo) const override;
+
+    bool IsSolutionStepFinalized() const;
+
+    bool IsSolutionStepInitialized() const;
```

Comment:
And the same thing here:
```suggestion
    bool IsSolutionStepInitialized() const;
    bool IsSolutionStepFinalized() const;
```

Reply:
Done